azurerm/resource_arm_subnet.go

```diff
@@ -319,11 +319,15 @@ func expandSubnetServiceEndpoints(d *schema.ResourceData) []network.ServiceEndpo
 	serviceEndpoints := d.Get("service_endpoints").([]interface{})
 	enpoints := make([]network.ServiceEndpointPropertiesFormat, 0)
 
-	for _, serviceEndpointsRaw := range serviceEndpoints {
-		data := serviceEndpointsRaw.(string)
+	for _, svcEndpointRaw := range serviceEndpoints {
+		if svcEndpointRaw == nil {
```

Comment:
As this is an `interface{}` not pointer shouldn't this be checked by:
```go
if service, ok := serviceEndpoint.(string); ok {
  endpoints = append(endpoints, network.ServiceEndpointPropertiesFormat{
    Service: &service,
  }
}
```

Also would be nice to fix the mis spelling of `endpoints` and not use a continue here.